src/components/slide-toggle/slide-toggle.ts

```diff
@@ -0,0 +1,130 @@
+import {
+  Component,
+  ElementRef,
+  Renderer,
+  forwardRef,
+  ChangeDetectionStrategy,
+  Input,
+  Output,
+  EventEmitter
+} from '@angular/core';
+import {
+  ControlValueAccessor,
+  NG_VALUE_ACCESSOR
+} from '@angular/common';
+import { BooleanFieldValue } from '../../core/annotations/field-value';
+import { Observable } from 'rxjs/Observable';
+
+export const MD_SLIDE_TOGGLE_VALUE_ACCESSOR: any = {
+  provide: NG_VALUE_ACCESSOR,
+  useExisting: forwardRef(() => MdSlideToggle),
+  multi: true
+};
+
+// Increasing integer for generating unique ids for slide-toggle components.
+let nextId = 0;
+
+@Component({
+  selector: 'md-slide-toggle',
+  host: {
+    '[class.md-checked]': 'checked',
+    '[class.md-disabled]': 'disabled',
+  },
+  templateUrl: './components/slide-toggle/slide-toggle.html',
+  styleUrls: ['./components/slide-toggle/slide-toggle.css'],
+  providers: [MD_SLIDE_TOGGLE_VALUE_ACCESSOR],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class MdSlideToggle implements ControlValueAccessor {
+
+  private onChange = (_: any) => {};
+  private onTouched = () => {};
+
+  // A unique id for the slide-toggle. By default the id is auto-generated.
+  private _uniqueId = `md-slide-toggle-${++nextId}`;
+  private _checked: boolean = false;
+  private _color: string;
+
+  @Input() @BooleanFieldValue() disabled: boolean = false;
+  @Input() name: string;
+  @Input() id: string = this._uniqueId;
+  @Input() tabIndex: number = 0;
+  @Input() ariaLabel: string = '';
+  @Input() ariaLabelledby: string = null;
+
+  @Output('change') private _change: EventEmitter<boolean> = new EventEmitter<boolean>();
+  change: Observable<boolean> = this._change.asObservable();
+
+  // Returns the unique id for the visual hidden input.
+  getInputId = () => `${this.id || this._uniqueId}-input`;
+
+  constructor(private elementRef: ElementRef,
+              private renderer: Renderer) {
+  }
+
+  /**
+   * The onChangeEvent method will be also called on click.
+   * This is because everything for the slide-toggle is wrapped inside of a label,
+   * which triggers a onChange event on click.
+   * @internal
+   */
+  onChangeEvent() {
+    if (!this.disabled) {
+      this.toggle();
+    }
+  }
+
+  /** @internal */
+  writeValue(value: any): void {
+    this.checked = value;
+  }
+
+  /** @internal */
+  registerOnChange(fn: any): void {
+    this.onChange = fn;
+  }
+
+  /** @internal */
+  registerOnTouched(fn: any): void {
+    this.onTouched = fn;
+  }
+
+  @Input()
+  get checked() {
+    return !!this._checked;
+  }
+
+  set checked(value) {
+    if (this.checked !== !!value) {
+      this._checked = value;
+      this.onChange(this._checked);
+      this._change.emit(this._checked);

```

Comment:
I tested that, and it is now covered by tests. But it seems like this is a more specific problem with the radio component. I cannot reproduce this issue with the `slide-toggle` component.

But, I've noticed that as well, some commits before. But it seems like I fixed it anyway. 

eg. this check in the setter, does not trigger any change (if present) before init, because it's the same value.
```ts
    if (this.checked !== !!value) {

```